main/src/ca/uwaterloo/flix/language/phase/unification/set/SetFormula.scala

```diff
@@ -872,4 +872,146 @@ object SetFormula {
     }
   }
 
+  /**
+    * The Successive Variable Elimination algorithm.
+    *
+    * Returns the most-general unifier of the equation `f ~ empty` where `fvs` is the free
+    * variables in `f`. If there is no unifier then [[NoSolutionException]] is thrown.
+    *
+    * Eliminates variables one-by-one from the given list `fvs` recursively.
+    *
+    * If the recursively building formula is ever larger than `recSizeThreshold` then
+    * [[ComplexException]] is thrown. If `recSizeThreshold` is non-positive then there is no
+    * checking.
+    */
+  def successiveVariableElimination(f: SetFormula, fvs: List[Int], recSizeThreshold: Int): SetSubstitution = fvs match {
+    case Nil =>
+      // `emptyEquivalent` now treats the remaining constants as variables and goes through all
+      // assignments of those.
+      if (emptyEquivalent(f)) SetSubstitution.empty
+      else throw NoSolutionException()
+
+    case x :: xs =>
+      val fEmpty = SetSubstitution.singleton(x, Empty)(f)
+      val fUniv = SetSubstitution.singleton(x, Univ)(f)
```

Comment:
I think it would really be better to keep the short names used elsewhere. They match those used in the papers; so its possible to lookup and compare.

Reply:
hmm, i dont agree. but not enough to insist. ill change it back

Reply:
This I insist on. If I ever take it to Jaco he will not be able to read it :)